src/plugins/routing.js

```diff
@@ -12,148 +12,173 @@ import {
-    }
+function switchLocalePath (locale) {
+  const i18n = this.i18n
 
-    const name = this.getRouteBaseName()
-    if (!name) {
-      return ''
-    }
+  if (strategy === STRATEGIES.NO_PREFIX && locale && locale !== i18n.locale) {
+    // eslint-disable-next-line no-console
+    console.warn(`[${MODULE_NAME}] Passing non-current locale to switchLocalePath is unsupported when using no_prefix strategy`)
+  }
+
+  const name = this.getRouteBaseName()
+  if (!name) {
+    return ''
+  }
 
-    const { params, ...routeCopy } = this.$route
-    let langSwitchParams = {}
-    if (vuex && this.$store) {
-      langSwitchParams = this.$store.getters[`${vuex.moduleName}/localeRouteParams`](locale)
+  const route = this.route
+  const { params, ...routeCopy } = route
+  let langSwitchParams = {}
+  const store = this.store
+  if (vuex && store) {
+    langSwitchParams = store.getters[`${vuex.moduleName}/localeRouteParams`](locale)
+  }
+  const baseRoute = Object.assign({}, routeCopy, {
+    name,
+    params: {
+      ...params,
+      ...langSwitchParams,
+      0: params.pathMatch
     }
-    const baseRoute = Object.assign({}, routeCopy, {
-      name,
-      params: {
-        ...params,
-        ...langSwitchParams,
-        0: params.pathMatch
-      }
-    })
-    let path = this.localePath(baseRoute, locale)
-
-    // Handle different domains
-    if (this[i18nPath].differentDomains) {
-      const lang = this[i18nPath].locales.find(l => l[LOCALE_CODE_KEY] === locale)
-      if (lang && lang[LOCALE_DOMAIN_KEY]) {
-        let protocol
-        if (process.server) {
-          const isHTTPS = require('is-https')
-          const { req } = this.$options._parentVnode.ssrContext
-          protocol = isHTTPS(req) ? 'https' : 'http'
-        } else {
-          protocol = window.location.protocol.split(':')[0]
-        }
-        path = protocol + '://' + lang[LOCALE_DOMAIN_KEY] + path
+  })
+  let path = this.localePath(baseRoute, locale)
+
+  // Handle different domains
+  if (i18n.differentDomains) {
+    const lang = i18n.locales.find(l => l[LOCALE_CODE_KEY] === locale)
+    if (lang && lang[LOCALE_DOMAIN_KEY]) {
+      let protocol
+      if (process.server) {
+        const isHTTPS = require('is-https')
+        protocol = isHTTPS(this.req) ? 'https' : 'http'
       } else {
-        // eslint-disable-next-line no-console
-        console.warn(`[${MODULE_NAME}] Could not find domain name for locale ${locale}`)
+        protocol = window.location.protocol.split(':')[0]
       }
+      path = protocol + '://' + lang[LOCALE_DOMAIN_KEY] + path
+    } else {
+      // eslint-disable-next-line no-console
+      console.warn(`[${MODULE_NAME}] Could not find domain name for locale ${locale}`)
     }
-    return path
   }
+  return path
 }
 
-function getRouteBaseNameFactory (contextRoute) {
-  function givenOrCurrent (route) {
-    return route || this.$route
+function getRouteBaseName (route) {
+  route = this.route
```

Comment:
Should we remove the `route` argument from `getRouteBaseName`'s definition? It seems superfluous since we're assigning `route` here.

Reply:
Yes! :)